integration-builder/initialization.js

```diff
@@ -1,33 +1,69 @@
+/* eslint-disable no-undef */
+
+var groupIds = [],
+    consentMapping = {};
+
 var initialization = {
-    name: 'insertSDKNameHere',
-    initForwarder: function(forwarderSettings, userAttributes, userIdentities, processEvent, eventQueue) {
-        // forwarderSettings will have any SDK settings required to use your SDK (ie, API key)
-        if (!testMode) {
-            /* Load your Web SDK here using a variant of your snippet from your readme that your customers would generally put into their <head> tags
-               Generally, our integrations create script tags and append them to the <head>. Please follow the following format as a guide:
-            */
-
-            // var clientScript = document.createElement('script');
-            // clientScript.type = 'text/javascript';
-            // clientScript.async = true;
-            // clientScript.src = 'https://www.clientscript.com/static/clientSDK.js';   // <---- Update this to be your script
-            // (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(clientScript);
-            // clientScript.onload = function() {
-            //     if (clientSDKObject && eventQueue.length > 0) {
-            //         // Process any events that may have been queued up while forwarder was being initialized.
-            //         for (var i = 0; i < eventQueue.length; i++) {
-            //             processEvent(eventQueue[i]);
-            //         }
-            //          // now that each queued event is processed, we empty the eventQueue
-            //         eventQueue = [];
-            //     }
-            //    clientSDKObject.initialize(forwarderSettings.apiKey);
-            // };
-        } else {
-            // For testing, you should fill out this section in order to ensure any required initialization calls are made,
-            // clientSDKObject.initialize(forwarderSettings.apiKey)
+    name: 'OneTrust',
+    parseConsentGroupIds: function() {
+        groupIds = window.OnetrustActiveGroups ? window.OnetrustActiveGroups.split(',') : [];
+        return groupIds;
+    },
+    parseConsentMapping: function(forwarderSettings) {
+        consentMapping = parseConsentMapping(forwarderSettings.consentGroups) || {};
+        return consentMapping;
+    },
+    initForwarder: function(forwarderSettings) {
+        var self = this;
+        this.parseConsentMapping(forwarderSettings);
+        self.parseConsentGroupIds();
+        if (Optanon && Optanon.OnConsentChanged) {
+            Optanon.OnConsentChanged(function() {
+                self.parseConsentGroupIds();
+                self.createConsentEvents();
+            });
+        }
+    },
+    createConsentEvents: function () {
+        var location = window.location.href,
+            consentState = mParticle.Consent.createConsentState(),
```

Comment:
perhaps you should be querying for the user's current consent state and use that if not null, otherwise if purposes are set that are not configured via OneTrust then they'll be clobbered.